CMakeLists.txt

```
@@ -6,18 +6,23 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 
-# set llvm-libs install path
-if(NOT DEFINED LLVM_INSTALL_PATH OR LLVM_INSTALL_PATH STREQUAL "")
-    message(FATAL_ERROR "Error: The variable LLVM_INSTALL_PATH is not set. Please specify it with -DLLVM_INSTALL_PATH=<value>.")
-endif()
+include(cmake/llvm_setup.cmake)
+
+setup_llvm()
+
 get_filename_component(LLVM_INSTALL_PATH "${LLVM_INSTALL_PATH}" ABSOLUTE)
 
 if(NOT EXISTS "${LLVM_INSTALL_PATH}")
     message(FATAL_ERROR "Error: The specified LLVM_INSTALL_PATH does not exist: ${LLVM_INSTALL_PATH}")
 endif()
 message(STATUS "Found llvm-libs ${LLVM_INSTALL_PATH}")
 
-set(CMAKE_PREFIX_PATH "${LLVM_INSTALL_PATH}")
+if (EXISTS "${LLVM_CMAKE_DIR}")
+    set(CMAKE_PREFIX_PATH "${LLVM_CMAKE_DIR}")
+else()
+    set(CMAKE_PREFIX_PATH "${LLVM_INSTALL_PATH}")
+endif()
```

If I understand correctly:
```suggestion
if (EXISTS "${LLVM_CMAKE_DIR}")
    list(APPEND CMAKE_MODULE_PATH "${LLVM_CMAKE_DIR}")
else()
    list(APPEND CMAKE_PREFIX_PATH "${LLVM_INSTALL_PATH}")
endif()
```